EpsilonAuxBMS/AuxBMSTests.adoc

```diff
@@ -0,0 +1,235 @@
+| Min Cell Voltage In Range |  0x721  | 0-2 |  1000ms |
+|===
+
+= Contactor Tests
+
+== Precharge State Test
+----
+Elapsed TIME_FRAME{
+    	Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX])
+}
+----
+
+== Aux Voltage Test 
+----
+Input Voltage From Voltage Source
+Elapsed TIME_FRAME{
+    	Expect (VOLTAGE_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == Input Voltage)
+}
+----
+
+== Discharge Contactor State Test
+----
+//Start Test
+Within TIME_FRAME{
+	Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX])
+}
+----
+
+== Charge Contactor State Test
+----
+//Start Test
+Within TIME_FRAME{
+	Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX])
+}
+----
+
+== Common Contactor State Test
+----
+//Start Test
+Within TIME_FRAME{
+	Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX])
+}
+----
+
+== Contactor Error Test
+----
+//Remove A Contactor In DONE State
+Within TIME_FRAME{
+	Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b000)
+}
+----
+
+== All Contactor On Test
+----
+//Start Test
+Within TIME_FRAME{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b001) 
+        COMMON LED ON
+        CHARGE LED OFF
+        DISCHARGE LED OFF
+}
+Within TIME_FRAME{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b011) 
+        COMMON LED ON
+        CHARGE LED ON
+        DISCHARGE LED OFF
+}
+Within TIME_FRAME{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b111) 
+        COMMON LED ON
+        CHARGE LED ON
+        DISCHARGE LED ON
+}
+Within TIME_FRAME{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b111) 
+   	    COMMON LED ON
+   	    CHARGE LED ON
+   	    DISCHARGE LED ON
+}
+----
+
+== Common and Discharge Cycle Test
+----
+//Start Test With Only Common Contactor Connected
+Within TIME_FRAME{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b001) 
+        COMMON LED ON
+        CHARGE LED OFF
+        DISCHARGE LED OFF
+}
+Within TIME_FRAME{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b001) 
+        COMMON LED ON
+        CHARGE LED OFF
+        DISCHARGE LED OFF
+}
+Within TIME_FRAME{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b001) 
+        COMMON LED ON
+        CHARGE LED OFF
+        DISCHARGE LED OFF
+}
+Within TIME_FRAME{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b001) 
+		COMMON LED ON
+        CHARGE LED OFF
+        DISCHARGE LED OFF
+}
+----
+
+== Discharge Off Test
+----
+//Start Discharge Test
+Within TIME_FRAME{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b101) 
+        COMMON LED ON
+        CHARGE LED ON
+        DISCHARGE LED OFF
+}
+----
+
+== Safety Test
+----
+//Start Safety Test
+Within TIME_FRAME{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b000) 
+        COMMON LED OFF
+        CHARGE LED OFF
+        DISCHARGE LED OFF
+}
+----
+
+== Allow Charge Test
+----
+Input Voltage < MAX_CELL_VOLTAGE And Charge Enable On
```

Comment:
We should define the `MAX_CELL_VOLTAGE` and `MIN_CELL_VOLTAGE` somewhere, or even just have the actual values.